### PR TITLE
chore: cleanup eslint rules

### LINKED ISSUE #1548: chore: cleanup eslint rules
# Description

We have plenty of eslint activated that make no sense for our library and get disabled manually all over the place.

* @angular-eslint/directive-selector
* @angular-eslint/directive-class-suffix
* @angular-eslint/component-selector
* @angular-eslint/component-class-suffix
* @angular-eslint/no-input-rename
* @typescript-eslint/member-ordering
* @typescript-eslint/no-non-null-assertion -> I would rather suggest finally going for strict mode

On the other side we are missing some very useful eslint rules:

* [simple-import-sort](https://www.npmjs.com/package/eslint-plugin-simple-import-sort)/imports
* [simple-import-sort](https://www.npmjs.com/package/eslint-plugin-simple-import-sort)/exports
* [unused-imports](https://www.npmjs.com/package/eslint-plugin-unused-imports)

# Tasks

* [ ] clean up eslint rules and unify them. This may also affect the imported plugins
* [ ] remove `// eslint-disable-next-line` comments where possible
* [ ] add new rules as suggested
* [ ] run `eslint --fix` for all files

### PR DESCRIPTION
**Pull request by naaajii**
closes #1548

-----------

`@angular-eslint/directive-selector` is mainly disabled in template and I believe changing the directive selector in template would be a breaking change.
`@angular-eslint/directive-class-suffix` this will require us to change `RxIf` to `RxIfDirective` and so on with other directives or we can just remove disabled rules and turn off the rule.
`@angular-eslint/no-input-rename` this rule is just simply disabled so I am not sure how to go with this one

I'm sorry for leaving this hanging last time cause I'm just crushed with uni and work so my apologies, hopefully we will be able to close this off this time!

### CODE REVIEW

**Review by edbzn — APPROVED** [review]
Looks good, thanks! I would just remove the scope `schematics` from the commit message but we can do it when we squash.

**Comment by naaajii** [comment]
should I also go ahead with naming the directives in template? 

`rx-angular\libs\template\unpatch\src\lib\viewport-prio.directive.ts`
```diff
- @Directive({ selector: '[viewport-prio]' })
+ @Directive({ selector: '[viewport-prio], [rxViewportPrio]' })
```

`rx-angular\libs\template\experimental\virtual-scrolling\src\lib\virtual-scroll-window.directive.ts`
```diff
- @Directive({ selector: 'rx-virtual-scroll-viewport[scrollWindow]' })
+ @Directive({ selector: '[rx-virtual-scroll-viewport[scrollWindow]], [rx-virtual-scroll-viewport[rxScrollWindow]]' })
```

`rx-angular\libs\template\experimental\virtual-scrolling\src\lib\scroll-strategies\autosize-virtual-scroll-strategy.ts`
```diff
- @Directive({ selector: 'rx-virtual-scroll-viewport[autosize]' })
+ @Directive({ selector: '[rx-virtual-scroll-viewport[autosize]], [rx-virtual-scroll-viewport[rxAutosize]]' })
```

`rx-angular\libs\template\experimental\virtual-scrolling\src\lib\scroll-strategies\dynamic-size-virtual-scroll-strategy.ts`
```diff
- @Directive({ selector: 'rx-virtual-scroll-viewport[dynamic]' })
+ @Directive({ selector: '[rx-virtual-scroll-viewport[dynamic]], [rx-virtual-scroll-viewport[rxDynamic]]' })
```

`rx-angular\libs\template\experimental\virtual-scrolling\src\lib\scroll-strategies\fixed-size-virtual-scroll-strategy.ts`
```diff
- @Directive({ selector: 'rx-virtual-scroll-viewport[itemSize]' })
+ @Directive({ selector: '[rx-virtual-scroll-viewport[itemSize]], [rx-virtual-scroll-viewport[rxItemSize]]' })
```

`rx-angular\libs\template\unpatch\src\lib\unpatch.directive.ts`
```diff
- @Directive({ selector: '[unpatch]' })
+ @Directive({ selector: '[unpatch], [rxUnpatch]' })
```

or we should just go with new selectors with rx prefix & camelcase then replace the existing selectors with new one? (would be quite breaking wouldn't it 😅)

I also think we should do different PRs for `@angular-eslint/directive-selector` (with a migration) & `@angular-eslint/directive-class-suffix`.

**Comment by hoebbelsB** [comment]
> should I also go ahead with naming the directives in template?
> 
> `rx-angular\libs\template\unpatch\src\lib\viewport-prio.directive.ts`
> 
> ```diff
> - @Directive({ selector: '[viewport-prio]' })
> + @Directive({ selector: '[viewport-prio], [rxViewportPrio]' })
> ```
> 
> `rx-angular\libs\template\experimental\virtual-scrolling\src\lib\virtual-scroll-window.directive.ts`
> 
> ```diff
> - @Directive({ selector: 'rx-virtual-scroll-viewport[scrollWindow]' })
> + @Directive({ selector: '[rx-virtual-scroll-viewport[scrollWindow]], [rx-virtual-scroll-viewport[rxScrollWindow]]' })
> ```
> 
> `rx-angular\libs\template\experimental\virtual-scrolling\src\lib\scroll-strategies\autosize-virtual-scroll-strategy.ts`
> 
> ```diff
> - @Directive({ selector: 'rx-virtual-scroll-viewport[autosize]' })
> + @Directive({ selector: '[rx-virtual-scroll-viewport[autosize]], [rx-virtual-scroll-viewport[rxAutosize]]' })
> ```
> 
> `rx-angular\libs\template\experimental\virtual-scrolling\src\lib\scroll-strategies\dynamic-size-virtual-scroll-strategy.ts`
> 
> ```diff
> - @Directive({ selector: 'rx-virtual-scroll-viewport[dynamic]' })
> + @Directive({ selector: '[rx-virtual-scroll-viewport[dynamic]], [rx-virtual-scroll-viewport[rxDynamic]]' })
> ```
> 
> `rx-angular\libs\template\experimental\virtual-scrolling\src\lib\scroll-strategies\fixed-size-virtual-scroll-strategy.ts`
> 
> ```diff
> - @Directive({ selector: 'rx-virtual-scroll-viewport[itemSize]' })
> + @Directive({ selector: '[rx-virtual-scroll-viewport[itemSize]], [rx-virtual-scroll-viewport[rxItemSize]]' })
> ```


The virtual-scrolling api should not be touched at all. The reasoning is that it should act as a drop-in for the cdk alternative, which has the very same selectors.

> `rx-angular\libs\template\unpatch\src\lib\unpatch.directive.ts`
> 
> ```diff
> - @Directive({ selector: '[unpatch]' })
> + @Directive({ selector: '[unpatch], [rxUnpatch]' })
> ```
> 
> or we should just go with new selectors with rx prefix & camelcase then replace the existing selectors with new one? (would be quite breaking wouldn't it 😅)

I would not introduce `rxUnpatch` here, `unpatch` is fine as is imo.

> 
> I also think we should do different PRs for `@angular-eslint/directive-selector` (with a migration) & `@angular-eslint/directive-class-suffix`.

**Comment by naaajii** [comment]
I think we can then just remove `@angular-eslint/directive-selector` and disabled rules from `template`?

`@angular-eslint/directive-class-suffix` should we add `Directive` suffix to `RxVirtualFor`, `AutoSizeVirtualScrollStrategy`, `DynamicSizeVirtualScrollStrategy`, `FixedSizeVirtualScrollStrategy`, `RxFor`, `RxIf`, `RxLet`, `RxUnpatch` or just turn off the rule in template?

lastly more or less same for `@angular-eslint/no-input-rename`, which will require us to either keep the rule disabled in template (as it is ) or change API to something like (which we aren't aiming for?):

```diff
+ @Input() rxLetParent = this.strategyProvider.config.parent;
- @Input('rxLetParent') renderParent = this.strategyProvider.config.parent;
```

**Comment by hoebbelsB** [comment]
> I think we can then just remove `@angular-eslint/directive-selector` and disabled rules from `template`?
> 
> `@angular-eslint/directive-class-suffix` should we add `Directive` suffix to `RxVirtualFor`, `AutoSizeVirtualScrollStrategy`, `DynamicSizeVirtualScrollStrategy`, `FixedSizeVirtualScrollStrategy`, `RxFor`, `RxIf`, `RxLet`, `RxUnpatch` or just turn off the rule in template?
> 
> lastly more or less same for `@angular-eslint/no-input-rename`, which will require us to either keep the rule disabled in template (as it is ) or change API to something like (which we aren't aiming for?):
> 
> ```diff
> + @Input() rxLetParent = this.strategyProvider.config.parent;
> - @Input('rxLetParent') renderParent = this.strategyProvider.config.parent;
> ```

yes, please disable all three of them

**Comment by naaajii** [comment]
that should do it, one thing I just noticed in changes

File: libs/template/unpatch/src/lib/unpatch.directive.ts [hunk 1]
```diff
@@ -116,12 +116,7 @@ export function unpatchEventListener(
  *
  * @publicApi
  */
-// eslint-disable-next-line @angular-eslint/directive-selector
 @Directive({ selector: '[unpatch]', standalone: true })
-/**
- * @todo: add prefix [rxUnpatch]
- */
```
Comment (naaajii) [inline libs/template/unpatch/src/lib/unpatch.directive.ts:123]: we aren't going to be making changes in API, so I thought we should remove the comment for todo.

**Comment by hoebbelsB** [comment]
> that should do it, one thing I just noticed in changes:
> 
> image
> 
> simple-import-sort or unused-imports rule isn't moving the comments around, so I will have to re-add them back to the lines they were associated with & we might wanna avoid running them to run in spec.ts files?

hm, that sucks. I mean it doesn't really matter in the end, but I think it's best to have them also applied in spec files

**Pull request merged**